### ISSUE TITLE
Composition of Lenses do not type properly

### ISSUE THREAD

**Issue by marcoemrich** [selfrefactor/rambda#714]
### Issue
Composing functions of type R.Lens do not provide a  proper Lens-Type.

Example: 

```typescript
const quantityLens = R.lensProp('quantity')
const composedLens = R.compose(quantityLens)
const result1 = R.view(quantityLens, {quantity: 5}) // this works
const result2 = R.view(composedLens, { quantity: 5 }) // this results in TS error (see screenshot)
```
![screenshot_lens-issue](https://github.com/selfrefactor/rambda/assets/447570/4bb37f34-2ef6-4a10-9045-59e5dfa04c1d)

### One-Liner to demo the issue

```typescript
const composedLens: R.Lens = R.compose(R.lensProp('foo')) 
```
Error:
```Property 'set' is missing in type '(obj: unknown) => unknown' but required in type 'Lens'.ts(2741)```

Maybe this is expected behavior. But it *does* work in Ramda - in that case, consider it a feature request instead of a bug.

### Rambda version
8.6.0

### Node.js version
20.6.1

**Comment by selfrefactor** [comment]
I will sync with Ramda types and this should be fixed. I will post update as soon as I have it.

**Comment by selfrefactor** [comment]
I will need also your TS version. I did test with latest TS and this is what I see:

![2](https://github.com/selfrefactor/rambda/assets/2149294/7cc51752-e115-4a34-b581-3ed37132d396)

**Comment by marcoemrich** [comment]
TS-Version is 5.3.3

But isn't the import in the screenshot above coming from Ramda instead of Rambda?

**Comment by selfrefactor** [comment]
I wanted to first check how your example works in Ramda, so that I can aim for similar result

**Comment by marcoemrich** [comment]
The Ramda version needs the generics type info to work

```typescript
import * as R from 'ramda'

const quantityLens = R.lensProp<{ quantity: number }, 'quantity'>('quantity')
const composedLens = R.compose(quantityLens)
const result1 = R.view(quantityLens, { quantity: 5 })
const result2 = R.view(composedLens, { quantity: 5 })
```

**Comment by marcoemrich** [comment]
The fact that rambda does not need the annotation `<{ quantity: number }, 'quantity'>`, I would consider one of the big advantages of rambda over ramda

**Comment by selfrefactor** [comment]
I used your latest example with Rambda code. It is not great as `result1` is unknown, which led me to decision to change Rambda types with `ramda`. 

If you have better suggestion how to fix this without this big change, I am open to your feedback.

As I don't use lenses that much, I am not the best source for the correctness of their types, so I will do the fastest solution which is to sync.

**Comment by selfrefactor** [comment]
released with `9.0.0`
this issue is the reason for the breaking change as types of lenses are synced with `ramda`
